In [51]:
from google.colab import drive
drive.mount('/content/drive')



Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [52]:
 ! pip install -q kaggle

In [53]:
import os
os.environ['KAGGLE_CONFIG_DIR'] = "/content/drive/MyDrive/CAPSTONE/Husein/Recomendation_Learn"

In [54]:
!ls -a

.  ..  .config	drive  sample_data


In [67]:
import pandas as pd
import numpy as np

In [56]:
df_moviemetadata= pd.read_csv('/content/drive/MyDrive/CAPSTONE/Husein/Recomendation_Learn/movies_metadata.csv')
df_moviemetadata.head(2)

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2718: DtypeWarning: Columns (10) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


adult  ... vote_count
0  False  ...     5415.0
1  False  ...     2413.0

[2 rows x 24 columns]

In [57]:
df_moviemetadata['genres'][0]

"[{'id': 16, 'name': 'Animation'}, {'id': 35, 'name': 'Comedy'}, {'id': 10751, 'name': 'Family'}]"

In [58]:
"""df_moviemetadata['belongs_to_collection'][0]
"""

"df_moviemetadata['belongs_to_collection'][0]\n"

In [59]:
"""df_moviemetadata['belongs_to_collection'][2]
"""

"df_moviemetadata['belongs_to_collection'][2]\n"

In [60]:
df_ratings=pd.read_csv('/content/drive/MyDrive/CAPSTONE/Husein/Recomendation_Learn/ratings.csv')
df_ratings.head(2)

userId  movieId  rating   timestamp
0       1      110     1.0  1425941529
1       1      147     4.5  1425942435

In [61]:
df_ratings['movieId']

0             110
1             147
2             858
3            1221
4            1246
            ...  
26024284    58559
26024285    60069
26024286    63082
26024287    64957
26024288    71878
Name: movieId, Length: 26024289, dtype: int64

In [62]:
from ast import literal_eval
# ini untuk menyambungkan kata kata yang dipisahkan
df_moviemetadata['genres'] = df_moviemetadata['genres'].fillna('[]').apply(literal_eval).apply(lambda x: [i['name'] for i in x] if isinstance(x, list) else [])
df_moviemetadata.head(2)

adult  ... vote_count
0  False  ...     5415.0
1  False  ...     2413.0

[2 rows x 24 columns]

In [63]:
"""df_moviemetadata['genres'] = df_moviemetadata['genres'].fillna('[]').apply(literal_eval).apply(lambda x: [i['name'] for i in x] if isinstance(x, list) else [])
df_moviemetadata.head(2)"""

"df_moviemetadata['genres'] = df_moviemetadata['genres'].fillna('[]').apply(literal_eval).apply(lambda x: [i['name'] for i in x] if isinstance(x, list) else [])\ndf_moviemetadata.head(2)"

In [64]:
movies=df_moviemetadata[['id','title','genres']]
movies['genres']=[''.join(st) for st in movies['genres']]
movies.head(2)

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


id      title                  genres
0   862  Toy Story   AnimationComedyFamily
1  8844    Jumanji  AdventureFantasyFamily

In [65]:
pd.crosstab(df_ratings['userId'],df_moviemetadata['id']).head(2)

id      100  10000  10001  100010  100017  ...  99946  9995  9997  99977  9999
userId                                     ...                                
1         0      0      0       0       0  ...      0     0     0      0     0
2         0      0      0       0       0  ...      0     0     0      0     0

[2 rows x 45436 columns]

`Encoder` : There are gaps in each user id and in each movie id, It's not sequential numbers like 0,1,2,3,4 .. etc it something like 862, 8844, 15602, .. etc, so we will use LabelEncoder() for making them in a sequential way like 0,1,2,3,4 .. etc.

In [68]:
from sklearn.preprocessing import LabelEncoder
encoder=LabelEncoder()
#df_ratings['user'] = encoder.fit_transform(df_ratings['userId'].values)
df_ratings['user'] = encoder.fit_transform(df_ratings['userId'].values)
n_users=df_ratings['user'].nunique()

df_ratings['movie'] = encoder.fit_transform(df_ratings['movieId'].values)
n_movies=df_ratings['movie'].nunique()

df_ratings['rating']=df_ratings['rating'].values.astype(np.float32)

min_rating = min(df_ratings['rating'])
max_rating = max(df_ratings['rating'])

n_users, n_movies, min_rating, max_rating

(270896, 45115, 0.5, 5.0)